Makefile

```diff
@@ -1,103 +1,113 @@
 
 checkstyle:
 	phpcs --standard=PSR2 ${SOURCE}
 
-checkstyle-ci: prepare
-	phpcs --report=checkstyle --report-file=${BASEDIR}/build/logs/checkstyle.xml --standard=PSR2 ${SOURCE} > /dev/null
+checkstyle-ci:
+	phpcs --report=checkstyle --report-file=./build/logs/checkstyle.xml --standard=PSR2 ${SOURCE} > /dev/null
 
-code-browser: prepare
-	phpcb --log ${BASEDIR}/build/logs --source ${SOURCE} --output ${BASEDIR}/build/code-browser
+code-browser:
+	phpcb --log ./build/logs --source ${SOURCE} --output ./build/code-browser
 
 # Copy paste detector
-cpd: prepare
-	phpcpd --log-pmd ${BASEDIR}/build/logs/pmd-cpd.xml ${SOURCE}
+cpd:
+	phpcpd --log-pmd ./build/logs/pmd-cpd.xml ${SOURCE}
 
-messdetector: prepare
-	phpmd ${SOURCE} text codesize,unusedcode,naming,design ${BASEDIR}/build/phpmd.xml
+messdetector:
+	phpmd ${SOURCE} text codesize,unusedcode,naming,design ./build/phpmd.xml
 
-messdetector-ci: prepare
-	phpmd ${SOURCE} xml codesize,unusedcode,naming,design --reportfile ${BASEDIR}/build/logs/pmd.xml
+messdetector-ci:
+	phpmd ${SOURCE} xml codesize,unusedcode,naming,design --reportfile ./build/logs/pmd.xml
 
-dependencies: prepare
-	pdepend --jdepend-xml=${BASEDIR}/build/logs/jdepend.xml \
-		--jdepend-chart=${BASEDIR}/build/pdepend/dependencies.svg \
-		--overview-pyramid=${BASEDIR}/build/pdepend/overview-pyramid.svg \
+dependencies:
+	pdepend --jdepend-xml=./build/logs/jdepend.xml \
+		--jdepend-chart=./build/pdepend/dependencies.svg \
+		--overview-pyramid=./build/pdepend/overview-pyramid.svg \
 		${SOURCE}
 
-phpunit: prepare
-	phpunit -c test/ --coverage-clover build/coverage/unit-coverage.xml --group unit
-	phpunit -c test/ --coverage-clover build/coverage/functional-coverage.xml --group functional
-	phpunit -c test/ --coverage-clover build/coverage/shutdown-coverage.xml --group shutdown
+phpunit:
+	-phpunit -c test/ --coverage-clover build/coverage/unit-coverage.xml --group unit
+	-phpunit -c test/ --coverage-clover build/coverage/functional-coverage.xml --group functional
+	-phpunit -c test/ --coverage-clover build/coverage/shutdown-coverage.xml --group shutdown
+	
+# Makes it easy to run a single test file. Example to run IndexTest.php: make test TEST="IndexTest.php"
+test:
+	${DOCKER_ENV} phpunit -c test/ test/lib/Elastica/Test/${TEST}
 
 doc: prepare
-	phpdoc run -d lib/ -t build/docs
+	${DOCKER} phpdoc run -d lib/ -t build/docs
 
 # Uses the preconfigured standards in .php_cs
 lint:
-	php-cs-fixer fix
+	${DOCKER} php-cs-fixer fix
 
 syntax-check:
 	php -lf ${SOURCE} **/*.php
-	php -lf ${BASEDIR}/test **/*.php
-
+	php -lf ./test **/*.php
 
-loc:
+loc: 
 	cloc --by-file --xml --exclude-dir=build -out=build/cloc.xml .
 
 phploc:
-	phploc --log-csv $(BASEDIR)/build/logs/phploc.csv $(SOURCE)
+	phploc --log-csv ./build/logs/phploc.csv ${SOURCE}
+
+# Handling virtual environment
+
+build:
+	docker-compose build
+
+setup: build
+	docker-compose scale elasticsearch=3
+	docker-compose run elasticsearch chmod -R 777 /mount/
```

Comment:
@im-denisenko I'm really unhappy about this line. Not the 777 but that I have to change the access rights after the images are running. What I need is a shared data container where all elasticsearch instances have write access. I tried to do it directly in the data image https://github.com/ruflin/Elastica/pull/901/files#diff-6ddc070b659c249d339814ba724740edR10 or also in the elasticsearch https://github.com/ruflin/Elastica/pull/901/files#diff-66fcadff5e0bc20f3becadf59df62f5bR28 But only the setup part works. I'm quite sure there is just a small command that can be used to get it working. Do you have any experience here?